server/server/src/io/log_manager.rs

```diff
@@ -59,37 +60,29 @@ impl LogManager {
     }
     /// Removes log file (if exists) and creates a new one.
     /// Returns path to the new log file.
-    fn recreate_log_file() -> String {
-        let mut log_path = String::from(std::env::current_dir().unwrap().to_str().unwrap());
+    fn recreate_log_file() -> PathBuf {
+        let mut log_path;
 
-        // Check ending.
-        #[cfg(target_os = "linux")]
+        #[cfg(any(unix, windows))]
         {
-            if !log_path.ends_with('/') {
-                log_path += "/";
-            }
-        }
-        #[cfg(target_os = "windows")]
-        {
-            if !log_path.ends_with('\\') {
-                log_path += "\\";
-            }
-        }
+            let user_dirs = UserDirs::new().expect(&format!(
+                "An error occurred at [{}, {}]: can't read user dirs.",
+                file!(),
+                line!(),
+            ));
 
-        log_path += LOG_DIR;
-
-        // Ending.
-        #[cfg(target_os = "linux")]
-        {
-            log_path += "/";
+            log_path = user_dirs.document_dir;
         }
-        #[cfg(target_os = "windows")]
+
+        #[cfg(not(any(unix, windows)))]
         {
-            log_path += "\\";
+            log_path = std::env::current_dir().unwrap();
```

Comment:
For consistency, I think we should show a compile error here, thus, when we port the project to MacOS we won't forget to put logs in the correct directory.

Reply:
Yes, MacOS falls under `unix` family. So the only OSes not included are mobile OSes and maybe some obscure ones.
But I'll change this to a compilation error, no problem.